// @aws-sdk/client-resource-groups-tagging-api/dist/es/models/models_0.js
import { __assign } from "tslib";
export var ComplianceDetails;
(function (ComplianceDetails) {
    ComplianceDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ComplianceDetails || (ComplianceDetails = {}));
export var ConcurrentModificationException;
(function (ConcurrentModificationException) {
    ConcurrentModificationException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConcurrentModificationException || (ConcurrentModificationException = {}));
export var ConstraintViolationException;
(function (ConstraintViolationException) {
    ConstraintViolationException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConstraintViolationException || (ConstraintViolationException = {}));
export var DescribeReportCreationInput;
(function (DescribeReportCreationInput) {
    DescribeReportCreationInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReportCreationInput || (DescribeReportCreationInput = {}));
export var DescribeReportCreationOutput;
(function (DescribeReportCreationOutput) {
    DescribeReportCreationOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReportCreationOutput || (DescribeReportCreationOutput = {}));
export var InternalServiceException;
(function (InternalServiceException) {
    InternalServiceException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InternalServiceException || (InternalServiceException = {}));
export var InvalidParameterException;
(function (InvalidParameterException) {
    InvalidParameterException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidParameterException || (InvalidParameterException = {}));
export var ThrottledException;
(function (ThrottledException) {
    ThrottledException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ThrottledException || (ThrottledException = {}));
export var ErrorCode;
(function (ErrorCode) {
    ErrorCode["INTERNAL_SERVICE_EXCEPTION"] = "InternalServiceException";
    ErrorCode["INVALID_PARAMETER_EXCEPTION"] = "InvalidParameterException";
})(ErrorCode || (ErrorCode = {}));
export var FailureInfo;
(function (FailureInfo) {
    FailureInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailureInfo || (FailureInfo = {}));
export var GroupByAttribute;
(function (GroupByAttribute) {
    GroupByAttribute["REGION"] = "REGION";
    GroupByAttribute["RESOURCE_TYPE"] = "RESOURCE_TYPE";
    GroupByAttribute["TARGET_ID"] = "TARGET_ID";
})(GroupByAttribute || (GroupByAttribute = {}));
export var GetComplianceSummaryInput;
(function (GetComplianceSummaryInput) {
    GetComplianceSummaryInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetComplianceSummaryInput || (GetComplianceSummaryInput = {}));
export var TargetIdType;
(function (TargetIdType) {
    TargetIdType["ACCOUNT"] = "ACCOUNT";
    TargetIdType["OU"] = "OU";
    TargetIdType["ROOT"] = "ROOT";
})(TargetIdType || (TargetIdType = {}));
export var Summary;
(function (Summary) {
    Summary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Summary || (Summary = {}));
export var GetComplianceSummaryOutput;
(function (GetComplianceSummaryOutput) {
    GetComplianceSummaryOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetComplianceSummaryOutput || (GetComplianceSummaryOutput = {}));
export var TagFilter;
(function (TagFilter) {
    TagFilter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagFilter || (TagFilter = {}));
export var GetResourcesInput;
(function (GetResourcesInput) {
    GetResourcesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourcesInput || (GetResourcesInput = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var ResourceTagMapping;
(function (ResourceTagMapping) {
    ResourceTagMapping.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceTagMapping || (ResourceTagMapping = {}));
export var GetResourcesOutput;
(function (GetResourcesOutput) {
    GetResourcesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourcesOutput || (GetResourcesOutput = {}));
export var PaginationTokenExpiredException;
(function (PaginationTokenExpiredException) {
    PaginationTokenExpiredException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PaginationTokenExpiredException || (PaginationTokenExpiredException = {}));
export var GetTagKeysInput;
(function (GetTagKeysInput) {
    GetTagKeysInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTagKeysInput || (GetTagKeysInput = {}));
export var GetTagKeysOutput;
(function (GetTagKeysOutput) {
    GetTagKeysOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTagKeysOutput || (GetTagKeysOutput = {}));
export var GetTagValuesInput;
(function (GetTagValuesInput) {
    GetTagValuesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTagValuesInput || (GetTagValuesInput = {}));
export var GetTagValuesOutput;
(function (GetTagValuesOutput) {
    GetTagValuesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTagValuesOutput || (GetTagValuesOutput = {}));
export var StartReportCreationInput;
(function (StartReportCreationInput) {
    StartReportCreationInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartReportCreationInput || (StartReportCreationInput = {}));
export var StartReportCreationOutput;
(function (StartReportCreationOutput) {
    StartReportCreationOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartReportCreationOutput || (StartReportCreationOutput = {}));
export var TagResourcesInput;
(function (TagResourcesInput) {
    TagResourcesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourcesInput || (TagResourcesInput = {}));
export var TagResourcesOutput;
(function (TagResourcesOutput) {
    TagResourcesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourcesOutput || (TagResourcesOutput = {}));
export var UntagResourcesInput;
(function (UntagResourcesInput) {
    UntagResourcesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourcesInput || (UntagResourcesInput = {}));
export var UntagResourcesOutput;
(function (UntagResourcesOutput) {
    UntagResourcesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourcesOutput || (UntagResourcesOutput = {}));
