// @aws-sdk/client-resource-groups-tagging-api/dist/es/commands/GetTagValuesCommand.js
import { __extends } from "tslib";
import { GetTagValuesInput, GetTagValuesOutput } from "../models/models_0";
import { deserializeAws_json1_1GetTagValuesCommand, serializeAws_json1_1GetTagValuesCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns all tag values for the specified key in the specified Region for the AWS
 *             account.</p>
 */
var GetTagValuesCommand = /** @class */ (function (_super) {
    __extends(GetTagValuesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function GetTagValuesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    GetTagValuesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ResourceGroupsTaggingAPIClient";
        var commandName = "GetTagValuesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTagValuesInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetTagValuesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTagValuesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetTagValuesCommand(input, context);
    };
    GetTagValuesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetTagValuesCommand(output, context);
    };
    return GetTagValuesCommand;
}($Command));
export { GetTagValuesCommand };
