// @aws-sdk/client-resource-groups-tagging-api-browser/model/operations/GetTagValues.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var GetTagValuesInput_1 = require("../shapes/GetTagValuesInput");
var GetTagValuesOutput_1 = require("../shapes/GetTagValuesOutput");
var InvalidParameterException_1 = require("../shapes/InvalidParameterException");
var ThrottledException_1 = require("../shapes/ThrottledException");
var InternalServiceException_1 = require("../shapes/InternalServiceException");
var PaginationTokenExpiredException_1 = require("../shapes/PaginationTokenExpiredException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.GetTagValues = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "GetTagValues",
    http: {
        method: "POST",
        requestUri: "/"
    },
    input: {
        shape: GetTagValuesInput_1.GetTagValuesInput
    },
    output: {
        shape: GetTagValuesOutput_1.GetTagValuesOutput
    },
    errors: [
        {
            shape: InvalidParameterException_1.InvalidParameterException
        },
        {
            shape: ThrottledException_1.ThrottledException
        },
        {
            shape: InternalServiceException_1.InternalServiceException
        },
        {
            shape: PaginationTokenExpiredException_1.PaginationTokenExpiredException
        }
    ]
};
