// @aws-sdk/client-resource-groups-tagging-api-browser/model/operations/TagResources.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagResourcesInput_1 = require("../shapes/TagResourcesInput");
var TagResourcesOutput_1 = require("../shapes/TagResourcesOutput");
var InvalidParameterException_1 = require("../shapes/InvalidParameterException");
var ThrottledException_1 = require("../shapes/ThrottledException");
var InternalServiceException_1 = require("../shapes/InternalServiceException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.TagResources = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "TagResources",
    http: {
        method: "POST",
        requestUri: "/"
    },
    input: {
        shape: TagResourcesInput_1.TagResourcesInput
    },
    output: {
        shape: TagResourcesOutput_1.TagResourcesOutput
    },
    errors: [
        {
            shape: InvalidParameterException_1.InvalidParameterException
        },
        {
            shape: ThrottledException_1.ThrottledException
        },
        {
            shape: InternalServiceException_1.InternalServiceException
        }
    ]
};
